source/agora/consensus/protocol/Nominator.d

```diff
@@ -466,6 +441,48 @@ extern(D):
             log.info("Rejected invalid envelope: {}", scpPrettify(&envelope));
     }
 
+    /***************************************************************************
+
+        Store the latest SCP sate for restore
+
+    ***************************************************************************/
+
+    public void storeLatestState () @safe nothrow
+    {
+        vector!SCPEnvelope envelopes;
+
+        () @trusted
+        {
+            if (this.scp.empty())
+                return;
+            envelopes = this.scp.getLatestMessagesSend(this.scp.getHighSlotIndex());
+        }();
+
+        // Clean the previous envelopes from the DB
+        this.scp_envelope_store.removeAll();
+
+        // Store the latest envelopes
+        foreach(const ref env; envelopes)
+            this.scp_envelope_store.add(env);
+    }
+
+    /***************************************************************************
+
+        Returns an instance of a SCPEnvelopeStore
```

Comment:
```suggestion
        Returns an instance of an SCPEnvelopeStore
```

(English is difficult, I know. :/
"A" and "An" is used based on pronunciation).